integrations/server_test.go

```diff
@@ -0,0 +1,519 @@
+	"github.com/influxdata/chronograf/server"
+)
+
+func TestServer(t *testing.T) {
+	type fields struct {
+		Organizations []chronograf.Organization
+		Users         []chronograf.User
+		Sources       []chronograf.Source
+		Servers       []chronograf.Server
+		Layouts       []chronograf.Layout
+		Dashboards    []chronograf.Dashboard
+	}
+	type args struct {
+		server    *server.Server
+		method    string
+		path      string
+		payload   interface{} // Expects this to be a json serializable struct
+		principal oauth2.Principal
+	}
+	type wants struct {
+		statusCode  int
+		contentType string
+		body        string
+	}
+
+	tests := []struct {
+		name    string
+		subName string
+		fields  fields
+		args    args
+		wants   wants
+	}{
+		{
+			name:    "GET /users",
+			subName: "User Not Found in the Default Organization",
+			fields:  fields{},
+			args: args{
+				server: &server.Server{
+					GithubClientID:     "not empty",
+					GithubClientSecret: "not empty",
+				},
+				method: "GET",
+				path:   "/chronograf/v1/users",
+				principal: oauth2.Principal{
+					Organization: "0",
+					Subject:      "billibob",
+					Issuer:       "github",
+					ExpiresAt:    time.Now().Add(10 * time.Second),
+					IssuedAt:     time.Now(),
+				},
+			},
+			wants: wants{
+				statusCode: 401,
+				body:       `{"code":401,"message":"User is not authorized"}`,
+			},
+		},
+		{
+			name:    "GET /users",
+			subName: "Single User in the Default Organization as SuperAdmin",
+			fields: fields{
+				Users: []chronograf.User{
+					{
+						ID:         1, // This is artificial, but should be reflective of the users actual ID
+						Name:       "billibob",
+						Provider:   "github",
+						Scheme:     "oauth2",
+						SuperAdmin: true,
+						Roles: []chronograf.Role{
+							{
+								Name:         "admin",
+								Organization: "0",
+							},
+						},
+					},
+				},
+			},
+			args: args{
+				server: &server.Server{
+					GithubClientID:     "not empty",
+					GithubClientSecret: "not empty",
+				},
+				method: "GET",
+				path:   "/chronograf/v1/users",
+				principal: oauth2.Principal{
+					Organization: "0",
+					Subject:      "billibob",
+					Issuer:       "github",
+					ExpiresAt:    time.Now().Add(10 * time.Second),
+					IssuedAt:     time.Now(),
+				},
+			},
+			wants: wants{
+				statusCode: 200,
+				body: `
+					{
+					  "links": {
+					    "self": "/chronograf/v1/users"
+					  },
+					  "users": [
+					    {
+					      "links": {
+					        "self": "/chronograf/v1/users/1"
+					      },
+					      "id": "1",
+					      "name": "billibob",
+					      "provider": "github",
+					      "scheme": "oauth2",
+					      "superAdmin": true,
+					      "roles": [
+					        {
+					          "name": "admin",
+					          "organization": "0"
+					        }
+					      ]
+					    }
+					  ]
+					}`,
+			},
+		},
+		{
+			name:    "GET /users",
+			subName: "Single User in the Default Organization as Admin",
```

Comment:
is this case even possible? i thought if there's only one user in the database, that user is guaranteed to be a superadmin? i understand that the test is for Admin (and Editor, Viewer, and Member below) being unauthorized, which this is good for, but the `fields` seem misleading..

Reply:
> i thought if there's only one user in the database, that user is guaranteed to be a superadmin?

This was before that change.

> is this case even possible?

You're right that it is not technically possible, but we have a little extra access to things that the rest of the users do not since we have raw access to the boltdb. It might be pretty easy to get into states that "arent possible". I wonder if there's a way to ensure that explicitly in these tests though. Might be a bit of cognitive overhead to have that always be the case. We should try to at the very least.

Reply:
Thoughts, @goller? I'd prefer to have any tests try as best as reasonably possible to reflect what's actually possible.